src/PrestaShopBundle/Controller/Admin/Sell/Order/InvoicesController.php

```diff
@@ -0,0 +1,223 @@
+     *
+     * @param Request $request
+
+     * @return RedirectResponse
+     */
+    public function generatePdfByStatusAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        if (!in_array(
+            $this->authorizationLevel($legacyController),
+            [
+                PageVoter::LEVEL_UPDATE,
+                PageVoter::LEVEL_CREATE,
+                PageVoter::LEVEL_DELETE,
+            ]
+        )) {
+            $this->addFlash(
+                'error',
+                $this->trans('You do not have permission to edit this', 'Admin.Notifications.Error')
+            );
+
+            return $this->redirectToRoute('admin_order_invoices');
+        }
+
+        $formHandler = $this->get('prestashop.admin.order.invoices.by_status.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
+                $this->flashErrors($errors);
+            } else {
+                $invoiceDataProvider = $this->get('prestashop.adapter.data_provider.order_invoice');
+                $invoiceCollection = [];
+
+                foreach ($data['generate_by_status']['order_states'] as $orderStateId) {
+                    // Put invoices for each selected status into one collection
+                    $invoiceCollection = array_merge(
+                        $invoiceCollection,
+                        $invoiceDataProvider->getByStatus($orderStateId)
+                    );
+                }
+
+                if (empty($invoiceCollection)) {
+                    $this->addFlash(
+                        'error',
+                        $this->trans('No invoice was found.', 'Admin.Orderscustomers.Notification')
+                    );
+                } else {
+                    // Generate PDF out of found invoices
+                    $invoiceGenerator = $this->get('prestashop.adapter.pdf_generator.invoice');
+                    $invoiceGenerator->generateInvoicesPDF($invoiceCollection);
+                }
+            }
+        }
+
+        return $this->redirectToRoute('admin_order_invoices');
+    }
+
+    /**
+     * Process the Invoice Options configuration form.
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processAction(Request $request)
+    {
+        $formHandler = $this->get('prestashop.admin.order.invoices.options.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
+                $this->flashErrors($errors);
+
+                return $this->redirectToRoute('admin_order_invoices');
```

Comment:
ping  @PierreRambaud, we can do something more pretty, but what? :)

Reply:
We can do that:
```php

    /**
     * Process the Invoice Options configuration form.
     *
     * @param Request $request
     *
     * @return RedirectResponse
     */
    public function processAction(Request $request)
    {
        $formHandler = $this->get('prestashop.admin.order.invoices.options.form_handler');
        $form = $formHandler->getForm();
        $form->handleRequest($request);
        if ($form->isValid() && $form->isSubmitted()) {
            $data = $form->getData();
            if ($errors = $formHandler->save($data)) {
                $this->flashErrors($errors);
            } else {
                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
            }
        }
        
        return $this->redirectToRoute('admin_order_invoices');
    }
```